korrecte-lib/src/linters/lints/alb_ingress_instance.rs

```diff
@@ -0,0 +1,187 @@
+use crate::linters::{Group, KubeObjectType, Lint, LintSpec};
+
+use crate::kube::ObjectRepository;
+use crate::reporting::Finding;
+use crate::reporting::Reporter;
+use k8s_openapi::api::core::v1::{ServiceSpec, ServiceStatus};
+use k8s_openapi::api::networking::v1beta1::{IngressSpec, IngressStatus};
+use kube::api::{KubeObject, Object};
+use std::collections::HashSet;
+
+/// **What it does:** Checks that all ALB ingresses are linked to service which have compatible types
+/// with the ingress. When ingress is configures with target-type `instance`, only `NodePort` and `LoadBalancer`
+/// types are allowed; when is configured as `ip`, only `ClusterIP` services are allowed.
+///
+/// **Why is this bad?** ALB ingress controller will fail to create the associated ALB if services
+/// have incompatible types.
+///
+/// **Known problems:**
+///
+/// **References**
+/// - https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/ingress/annotation/#target-type
+pub(crate) struct AlbIngressInstance<'a> {
+    object_repository: &'a dyn ObjectRepository,
+}
+
+impl<'a> AlbIngressInstance<'a> {
+    pub fn new(object_repository: &'a dyn ObjectRepository) -> Self {
+        AlbIngressInstance { object_repository }
+    }
+}
+
+impl<'a> Lint for AlbIngressInstance<'a> {
+    fn v1beta1_ingress(
+        &self,
+        ingress: &Object<IngressSpec, IngressStatus>,
+        reporter: &dyn Reporter,
+    ) {
+        let metadata = &ingress.meta().annotations;
+        let is_alb_ingress = metadata
+            .get("kubernetes.io/ingress.class")
+            .map(|class| class == "alb")
+            .unwrap_or(false);
+        if !is_alb_ingress {
+            return;
+        }
+
+        let services = Self::extract_service_names(ingress);
+        let ingress_type = IngressType::from(ingress);
+        let missconfigured_services = self.check_allowed_service_type(&ingress_type, services);
+
+        for service in missconfigured_services {
+            let finding = Finding::new(Self::spec(), ingress.meta().clone())
+                .add_metadata("service", service.meta().name.clone());
```

Comment:
why are you using clones to both Finding::new and add_metadata? Are we mutating that data anywhere?

Reply:
Note that, on line 52, I'm cloning the full metadata of the  `ingress` (so, the subject under linting).
On the `add_metadata`, I'm just cloning the service name. This will show the ingress which is misconfigured, and which service points to.